go/pools/smartconnpool/pool.go

```diff
@@ -503,8 +503,16 @@ func (pool *ConnPool[C]) connReopen(ctx context.Context, dbconn *Pooled[C], now
 		return err
 	}
 
-	dbconn.timeUsed.set(now)
+	if dbconn.Conn.Setting() != nil {
+		err = dbconn.Conn.ApplySetting(ctx, dbconn.Conn.Setting())
+		if err != nil {
+			dbconn.Close()
+			return err
+		}
+	}
+
```

Comment:
Was this just broken before? We were not applying settings on re-open?

Reply:
I checked, and in other places, we were resetting before calling reopen, so technically, it has been safe until now.

Reply:
This is called in few places. Most of them were handling it by using setting reset call.
It was not handled when `connection max lifetime` was achieved, in that case it would have returned to wrong stack.

Reply:
Do we need to backport just the fix for this into older release branches?